lib/internal/modules/esm/resolve.js

```diff
@@ -220,13 +210,16 @@ function finalizeResolution(resolved, base, preserveSymlinks) {
 
   const path = fileURLToPath(resolved);
 
-  const stats = tryStatSync(StringPrototypeEndsWith(path, '/') ?
+  const stats = internalModuleStat(StringPrototypeEndsWith(path, '/') ?
```

Comment:
This needs `toNamespacedPath()` too, and below (wherever `internalModuleStat` is used instead).